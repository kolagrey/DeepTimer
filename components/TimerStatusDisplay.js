import React from 'react';
import { SafeAreaView, Text } from 'react-native';
import PropTypes from 'prop-types';
import styles from '../styles';

const TimerStatusDisplay = (props) => {
    const { timerStatusText } = props;
    return (
        <SafeAreaView style={styles.middleContainer}>
            <Text style={styles.timerStatusText}>
                {timerStatusText}
            </Text>
        </SafeAreaView>
    );
};

TimerStatusDisplay.propTypes = {
    timerStatusText: PropTypes.string
};

export default TimerStatusDisplay;