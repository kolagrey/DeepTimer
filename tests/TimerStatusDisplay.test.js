import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import TimerStatusDisplay from '../components/TimerStatusDisplay';

describe('<TimerStatusDisplay/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<TimerStatusDisplay/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'RCTSafeAreaView'", () => {
        const tree = renderer
            .create(
                <TimerStatusDisplay />,
            ).toJSON();
        expect(tree.type).toBe('RCTSafeAreaView');
    });

    it('has a child', () => {
        const tree = renderer
            .create(
                <TimerStatusDisplay/>,
            ).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it('has a child of <Text> component', () => {
        const tree = renderer
            .create(
                <TimerStatusDisplay/>,
            ).toJSON();
        expect(tree.children[0].type).toBe('Text');
    });
});