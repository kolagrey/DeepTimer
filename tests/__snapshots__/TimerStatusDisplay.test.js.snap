// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimerStatusDisplay/> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
      }
    }
  />
</RCTSafeAreaView>
`;
